### PR TITLE
Add dotnet output type to oryx manifest file

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as USER STORY 1347854
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

**Review by arroyc — COMMENTED** [review]
nit: add a buildimage test which will try to build a dotnetcore app with buildimages such as full build image, jamstack image and test the content of the manifest file. you added detector tests which is good, but adding buildimage test sort of solidify the end to end flow.

File: src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs [hunk 1]
```diff
@@ -121,6 +121,9 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             manifestFileProperties[ManifestFilePropertyKeys.DotNetCoreSdkVersion]
                 = dotNetCorePlatformDetectorResult.SdkVersion;
 
+            string outputType = dotNetCorePlatformDetectorResult.OutputType;
+            AddOutputTypeOptionalField(manifestFileProperties, outputType);
```
Comment (arroyc) [inline src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs:125]: nit: you probably do not need a method to write outputtype to manifestfile unless you are writing an unit test to check how you are handling if outputtype is null/empty and so on

File: src/BuildScriptGenerator/DotNetCore/DotnetCoreConstants.cs [hunk 2]
```diff
@@ -32,6 +32,7 @@ public static class DotNetCoreConstants
         public const string ProjectSdkElementNameAttributeValueXPathExpression = "string(/Project/Sdk/@Name)";
         public const string TargetFrameworkElementXPathExpression = "/Project/PropertyGroup/TargetFramework";
         public const string AssemblyNameXPathExpression = "/Project/PropertyGroup/AssemblyName";
+        public const string OutputTypeAttributeValueXPathExpression = "/Project/PropertyGroup/OutputType";
```
Comment (arroyc) [inline src/BuildScriptGenerator/DotNetCore/DotnetCoreConstants.cs:35]: nit: rename it as OutputTypeXPathExpression

File: src/Detector/DotNetCore/DotnetCoreConstants.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ internal static class DotNetCoreConstants
         public const string ProjectSdkElementNameAttributeValueXPathExpression = "string(/Project/Sdk/@Name)";
         public const string TargetFrameworkElementXPathExpression = "/Project/PropertyGroup/TargetFramework";
         public const string AssemblyNameXPathExpression = "/Project/PropertyGroup/AssemblyName";
+        public const string OutputTypeElementXPathExpression = "/Project/PropertyGroup/OutputType";
```
Comment (arroyc) [inline src/Detector/DotNetCore/DotnetCoreConstants.cs:21]: nit:OutputTypeXPathExpression

File: tests/Detector.Tests/DotNetCore/SampleProjectFileContents.cs [hunk 4]
```diff
@@ -35,5 +35,13 @@ static class SampleProjectFileContents
             <RootNamespace>Microsoft.Oryx.BuildScriptGenerator.Tests</RootNamespace>
           </PropertyGroup>
         </Project>";
+
+        public const string ProjectFileWithOutputTypePlaceHolder = @"
```
Comment (arroyc) [inline tests/Detector.Tests/DotNetCore/SampleProjectFileContents.cs:39]: add few more cases like
ProjectFileWithoutOutputtypePlaceHolder
ProjectFileWithNullOutputtypePlaceHolder
ProjectFileWithEmptyOutputtypePlaceHolder

**Review by arroyc — CHANGES_REQUESTED** [review]

File: tests/Detector.Tests/DotNetCore/SampleProjectFileContents.cs (hunk 4 above)
Comment (william-msft) [reply tests/Detector.Tests/DotNetCore/SampleProjectFileContents.cs:39]: Do we think the checks in the unit tests would suffice? 
https://github.com/microsoft/Oryx/pull/1058/files/6ab2df967ffdcfe5aadda086254b06330b839c6f#diff-3c310ce937af62ff2c475327a2918ff29832e18bb3588baaabe08ed8a425042dR85 

It checks for null & invalid input. If there's something I might be missing, I'm happy to add them, just wanted to check with you first

Comment (william-msft) [reply tests/Detector.Tests/DotNetCore/SampleProjectFileContents.cs:39]: Added new unit test to detect when `OutputType ` is not present per discussion, the other 2 are already accounted for

**Review by arroyc — APPROVED** [review]

**Pull request merged**